Civi/Api4/Utils/FormattingUtil.php

```diff
@@ -126,11 +126,8 @@ public static function formatInputValue(&$value, ?string $fieldPath, array $fiel
     }
 
     // Special handling for 'current_user' and user lookups
-    if ($fk === 'Contact' && isset($value) && !is_numeric($value)) {
-      $value = \_civicrm_api3_resolve_contactID($value);
-      if ('unknown-user' === $value) {
-        throw new \CRM_Core_Exception("\"{$fieldSpec['name']}\" \"{$value}\" cannot be resolved to a contact ID", 2002, ['error_field' => $fieldSpec['name'], "type" => "integer"]);
-      }
+    if (is_string($fk) && CoreUtil::isContact($fk)) {
+      $value = self::resolveContactID($fieldSpec['name'], $value);
```

Comment:
This is both an abstraction and a fix for the case when "$fk" is set to the "entity" which when coming from eg. FormBuilder might be "Individual" (and probably when called directly using "Individual" pseudo API).